src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -187,30 +213,8 @@ void SotaUptaneClient::runForever(const std::shared_ptr<command::Channel> &comma
 
     try {
       if (command->variant == "GetUpdateRequests") {
+        getUpdateRequests();
```

Comment:
I like the direction, but then we sould probably catch exceptions inside getUpdateRequests(). Otherwise a non-fatal Uptane::Exception will be able to stop the polling process.

Reply:
Oh, right. We should put `schedulePool` in front of it, as it was before...